src/runners/alert_handler.py

```diff
@@ -57,54 +57,88 @@ def log_failure(ctx, alert, e):
         log.error("Failed to log alert creation failure", e)
 
 
-def get_new_alerts(connection):
+def get_new_alerts(ctx):
     get_alerts_query = f'SELECT * FROM {ALERTS_TABLE} WHERE ticket IS NULL AND suppressed=FALSE LIMIT 100'
-    results = connection.cursor().execute(get_alerts_query).fetchall()
+    results = ctx.cursor().execute(get_alerts_query).fetchall()
     print('Found', len(results), 'new alerts.')
     return results
 
 
-def record_ticket_id(connection, ticket_id, alert_id):
+def record_ticket_id(ctx, ticket_id, alert_id):
     query = f"UPDATE {ALERTS_TABLE} SET ticket='{ticket_id}' WHERE alert:ALERT_ID='{alert_id}'"
     print('Updating alert table:', query)
-    connection.cursor().execute(query)
+    ctx.cursor().execute(query)
+
+# We get a list of alerts that don't have tickets. For each alert, check the group_id of the alert; if there is no alert that has that group_id and a ticket_id, create a ticket for that alert.
+# if there is an alert with a matching group_id and a ticket_id, update the body of the ticket with the new alert instead.
 
 
 def main():
     ctx = db.connect_and_execute(f'USE DATABASE {DATABASE};')
     alerts = get_new_alerts(ctx)
-    print('Found', len(alerts), 'new alerts to handle.')
+    log.info('Found', len(alerts), 'new alerts to handle.')
 
     for row in alerts:
         try:
             alert = json.loads(row[0])
+            group_id = row[7]
```

Comment:
My initial run at this had the correlation_id as an element in the json; the problem I ran into is that pulling the json out and modifying it is easy, but inserting the JSON back into Snowflake is weirdly non-trivial; there are a BUNCH of cases where fields like EVENT_DATA are going to have both double and single quotes in them, and I couldn't figure out a good way to properly escape everything so that we wouldn't end up with parsing errors when we try to make the insert statement. I tried json.dumping the alert itself, I tried json.dumping the dangerous fields specifically, and I tried string.replace("'", "\'") on the dangerous fields, and none of those three options worked. If you know a better way to do it, I'm open to ideas.